docs/datadog_agent_profiles.md

```diff
@@ -0,0 +1,59 @@
+# DatadogAgentProfiles (beta)
+
+This feature was introduced in operator v1.5.0 and is currently in beta.
+
+## Overview
+
+DatadogAgentProfiles (DAPs), also known as profiles, can be created to override certain operator settings that were set in a DatadogAgent (DDA) on a subset of nodes. Currently, it is possible to override the node agent container resource settings using a DAP. While multiple DAPs can be applied to a cluster, each DAP must target a different subset of nodes so the DAPs do not conflict with each other. 
+
+Example:
+
+```yaml
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogAgentProfile
+metadata:
+  name: datadogagentprofile-sample
+spec:
+  profileAffinity:
+    profileNodeAffinity:
+      - key: kubernetes.io/os
+        operator: In
+        values:
+          - linux
+  config:
+    override:
+      nodeAgent:
+        containers:
+          agent:
+            resources:
+              requests:
+                cpu: 256m
+```
+
+The DAP spec has two main sections:
+* `profileAffinity` is used to target a subset of nodes. It acccepts a list of [NodeSelectorRequirements](https://pkg.go.dev/k8s.io/api/core/v1#NodeSelectorRequirement).
+* `config` defines the configuration to override in the DDA. It follows the configuration formatting of the operator's [DatadogAgentSpec](https://github.com/DataDog/datadog-operator/blob/98276c56ad824f81be6f75128d230d2c4eda4c0b/apis/datadoghq/v2alpha1/datadogagent_types.go#L28).
+
+When a DAP is applied, the operator creates a new DaemonSet for that profile using the name format `datadog-agent-with-profile-<namespace>-<name>`. Even if the operator is configured to use ExtendedDaemonSets, it will still create DaemonSets for any DAPs. It will also create a DaemonSet (or an ExtendedDaemonSet, if enabled) for a default profile. The default profile uses the same naming pattern that the DDA uses for node agents and applies to all nodes that are not targetted by a DAP.
```

Comment:
```suggestion
When a DAP is applied, the Operator creates a new DaemonSet for that profile using the name format `datadog-agent-with-profile-<namespace>-<name>`. Even if the Operator is configured to use ExtendedDaemonSets, it will still create DaemonSets for any DAPs. It will also create a DaemonSet (or an ExtendedDaemonSet, if enabled) for a default profile. The default profile uses the same naming pattern that the DDA uses for node agents and applies to all nodes that are not targeted by a DAP.
```